README.md
<p align="center">
    <img width="64" height="64" src="data/images/noyau.png">
</p>

<h1 align="center">
    <b>Noyau</b>
</h1>

<p align="center">
    A graphical utility for managing kernels on Ubuntu-based platforms.
</p>

<br />

<p align="center">
    <img width="720" height="570" src="data/screenshots/main-ubuntu.png">
</p>

<br />

[![CodeFactor](https://www.codefactor.io/repository/github/joshuadowding/noyau/badge)](https://www.codefactor.io/repository/github/joshuadowding/noyau)


<h2>Features</h2>
<ul>
    <li>Download and install kernel packages automatically.</li>
    <li>Purge all installed kernels except the running kernel.</li>
    <li>View kernel release changelogs with your default text editor.</li>
</ul>


<h2>Changes Since 18.9.1</h2>
<h3>19.10</h3>
<ul>
    <li>Refreshed user interface using the Gtk.HeaderBar.</li>
    <li>Transitioned to the meson build system.</li>
    <li>Merge the GUI and CLI program entry-points into one unified entry-point.</li>
    <li>Download over HTTPS instead of plain HTTP.</li>
    <li>Split toggles for displaying kernels older than 4.x and 5.x.</li>
    <li>Removed intrusive prompts and donation links.</li>
    <li>Removed scripted notifications.</li>
    <li>Serious code refactoring and clean-up.</li>
    <li>Fixed numerous build issues.</li>
</ul>


<h2>Confirmed Compatibility</h2>
<ul>
    <li>Ubuntu 20.04.x LTS</li>
    <li>Ubuntu 19.10</li>
    <li>Ubuntu 19.04</li>
    <li>Ubuntu 18.04.x LTS (inc. elementary OS 5.x)</li>
</ul>

<h2>Confirmed Incompatibility</h2>
<ul>
    <li>Ubuntu 16.04.x LTS</li>
</ul>


<h2>Source Code</h2>
<h3>Build Instructions</h3>

    sudo apt install valac meson libgtk-3-dev libgee-0.8-dev libjson-glib-dev libvte-2.91-dev libsoup2.4-dev libnotify-dev
    git clone https://github.com/joshuadowding/noyau.git && cd noyau
    meson build --prefix=/usr
    cd build && ninja
    sudo ninja install

<h3>Packaging Dependencies</h3>

    sudo apt install pbuilder debootstrap devscripts debhelper dh-make autotools-dev

<h3>Runtime Dependencies</h3>

    sudo apt install rsync aria2 aptitude curl



debian/changelog
com.github.joshuadowding.noyau (19.10) focal; urgency=low

  * Refreshed user interface using the Gtk.HeaderBar
  * Transitioned to the meson build system
  * Merge the GUI and CLI program entry-points into one unified entry-point
  * Download over HTTPS instead of plain HTTP
  * Split toggles for displaying kernels older than 4.x and 5.x
  * Removed intrusive prompts and donation links
  * Removed scripted notifications
  * Serious code refactoring and clean-up
  * Fixed numerous build issues

 -- Joshua Dowding <joshuadowding@outlook.com>  Sat, 18 Jan 2020 17:00:00 +0000



build-pbuilder.sh
#!/bin/bash

echo ""
echo "======================================================="
echo "build-pbuilder.sh - begin setup"
echo "======================================================="
echo ""

#sudo pbuilder --create --distribution xenial --architecture i386 --basetgz /var/cache/pbuilder/xenial-i386-base.tgz
#sudo pbuilder --create --distribution xenial --architecture amd64 --basetgz /var/cache/pbuilder/xenial-amd64-base.tgz

sudo pbuilder --create --distribution bionic --architecture i386 --basetgz /var/cache/pbuilder/bionic-i386-base.tgz
sudo pbuilder --create --distribution bionic --architecture amd64 --basetgz /var/cache/pbuilder/bionic-amd64-base.tgz

sudo pbuilder --create --distribution disco --architecture i386 --basetgz /var/cache/pbuilder/disco-i386-base.tgz
sudo pbuilder --create --distribution disco --architecture amd64 --basetgz /var/cache/pbuilder/disco-amd64-base.tgz

sudo pbuilder --create --distribution eoan --architecture i386 --basetgz /var/cache/pbuilder/eoan-i386-base.tgz
sudo pbuilder --create --distribution eoan --architecture amd64 --basetgz /var/cache/pbuilder/eoan-amd64-base.tgz

#sudo pbuilder --create --distribution focal --architecture i386 --basetgz /var/cache/pbuilder/focal-i386-base.tgz
sudo pbuilder --create --distribution focal --architecture amd64 --basetgz /var/cache/pbuilder/focal-amd64-base.tgz

## Based on: https://blog.packagecloud.io/eng/2015/05/18/building-deb-packages-with-pbuilder/


meson/build-source.sh
#!/bin/bash

backup=`pwd`
DIR="$( cd "$( dirname "$0" )" && pwd )"
cd "$DIR"

#../BUILD_CONFIG

echo ""
echo "=========================================================================="
echo " build-source.sh"
echo "=========================================================================="
echo ""

echo "app_name: $app_name"
echo "pkg_name: $pkg_name"
echo "--------------------------------------------------------------------------"

# clean build dir

rm -rfv /tmp/builds
mkdir -pv /tmp/builds

#ninja clean

mkdir -pv ../release/source

echo "--------------------------------------------------------------------------"

cd ..
cd ..

# build source package
dpkg-source --build ukuu

mv -vf $pkg_name*.dsc ukuu/release/source/
mv -vf $pkg_name*.tar.xz ukuu/release/source/

if [ $? -ne 0 ]; then cd "$backup"; echo "Failed"; exit 1; fi

echo "--------------------------------------------------------------------------"

# list files
ls -l ukuu/release/source

echo "-------------------------------------------------------------------------"

cd "$backup"



build-deb.sh
#!/bin/bash

backup=`pwd`
DIR="$( cd "$( dirname "$0" )" && pwd )"
cd "$DIR"

build_source_for_dist() {
    version=$1

    echo ""
    echo "======================================================="
    echo "build-deb.sh - build source - ${version}"
    echo "======================================================="
    echo ""

    debuild -i -us -uc -S

    cd ..

    mv -vf noyau_${version}.dsc noyau/release
    mv -vf noyau_${version}.tar.xz noyau/release
    mv -vf noyau_${version}_source.build noyau/release
    mv -vf noyau_${version}_source.buildinfo noyau/release
    mv -vf noyau_${version}_source.changes noyau/release

    if [ $? -ne 0 ]; then cd "$backup"; echo "Failure, again!"; exit 1; fi

    echo "--------------------------------------------------------------------------"
}

build_deb_for_dist() {
    dist=$1
    arch=$2
    version=$3

    echo ""
    echo "======================================================="
    echo "build-deb.sh - build deb - ${dist}, ${arch}, ${version}"
    echo "======================================================="
    echo ""

    cd noyau

    cd release
    mkdir -pv ${dist}/${arch}

    cd ..

    ls -l release/${dist}/${arch}

    mkdir -pv release/${dist}/${arch}

    echo "-------------------------------------------------------------------------"

    sudo -S pbuilder --build --distribution ${dist} --architecture ${arch} --basetgz /var/cache/pbuilder/${dist}-${arch}-base.tgz release/noyau_${version}.dsc

    if [ $? -ne 0 ]; then cd "$backup"; echo "Failure, again!"; exit 1; fi

    echo "--------------------------------------------------------------------------"

    sudo cp -pv /var/cache/pbuilder/result/noyau_${version}_${arch}.deb release/${dist}/${arch}
    sudo cp -pv /var/cache/pbuilder/result/noyau_${version}_${arch}.changes release/${dist}/${arch}

    if [ "${dist}" != "xenial" ]; then
        sudo cp -pv /var/cache/pbuilder/result/noyau_${version}_${arch}.buildinfo release/${dist}/${arch}
        sudo cp -pv /var/cache/pbuilder/result/noyau-dbgsym_${version}_${arch}.ddeb release/${dist}/${arch}
    fi

    if [ $? -ne 0 ]; then cd "$backup"; echo "Failure, again!"; exit 1; fi

    sudo rm -r /var/cache/pbuilder/result/*

    echo "--------------------------------------------------------------------------"
}

echo ""
echo "======================================================="
echo "build-deb.sh - begin build"
echo "======================================================="
echo ""

mkdir release

build_source_for_dist 19.10-beta~ppa

#build_deb_for_dist xenial i386 19.10-beta~ppa  #16.04.x LTS (not supported)
#build_deb_for_dist xenial amd64 19.10-beta~ppa #16.04.x LTS (not supported)

#build_deb_for_dist bionic i386 19.10-beta~ppa #18.04.x LTS
build_deb_for_dist bionic amd64 19.10-beta~ppa #18.04.x LTS

#build_deb_for_dist disco i386 19.10-beta~ppa #19.04
#build_deb_for_dist disco amd64 19.10-beta~ppa #19.04

#build_deb_for_dist eoan i386 19.10-beta~ppa #19.10
#build_deb_for_dist eoan amd64 19.10-beta~ppa #19.10

#build_deb_for_dist focal i386 19.10-beta~ppa #20.04.x LTS
build_deb_for_dist focal amd64 19.10-beta~ppa #20.04.x LTS

cd "$backup"
